broadcast/src/lib.rs

```diff
@@ -14,6 +15,14 @@ pub trait Broadcaster<Cfg: Config>: Clone + Send + 'static {
     /// - deserialized upon reception
     type Message: Codec<Cfg> + Clone + Send + 'static;
 
+    /// Response is the type of data that is returned once the message is broadcasted.
+    ///
+    /// It may also indicate the success or failure of the broadcast attempt.
+    type Response: Send + 'static;
+
     /// Attempt to broadcast a message to the network.
-    fn broadcast(&mut self, message: Self::Message) -> impl Future<Output = ()> + Send;
+    fn broadcast(
+        &mut self,
+        message: Self::Message,
+    ) -> impl Future<Output = oneshot::Receiver<Self::Response>> + Send;
```

Comment:
I like this pattern of returning the waiter so that folks that don't care about delivery metadata aren't forced to wait for it.